code/datums/components/uplink.dm

```diff
@@ -168,13 +168,15 @@
 	if(trim(lowertext(new_ringtone)) != trim(lowertext(unlock_code)))
 		return
 	locked = FALSE
-	tgui_interact(user)
+	if(user)
+		tgui_interact(user)
```

Comment:
You put these here so you can invoke these events with a null user when a PDA is infected. I think you should just set `locked = FALSE` instead of invoking the events.

Reply:
This actually opens the uplink for the user when the code is input

Reply:
Done.